### PR TITLE
Gutenberg Blocks

### PR DESCRIPTION
**Pull request by NSHouseCat**
* Omnibar
* IDX Broker:
  * Widgets
  * Wrapper Tags
* IMPress:
  * Carousel
  * Showcase
  * City Links
  * Lead Signup
  * Lead Login

### CODE REVIEW

File: idx/blocks/register-blocks.php [hunk 1]
```diff
@@ -0,0 +1,739 @@
...
+
+	/**
+	 * Register_Idx_Category function.
+	 *
+	 * @access public
+	 * @return mixed
+	 */
+	public function register_idx_category( $categories, $post ) {
+		if ( $post->post_type !== 'post' ) {
+				return $categories;
```
Comment (ErikIDX) [inline idx/blocks/register-blocks.php]: Doubled indented

File: idx/blocks/register-blocks.php [hunk 2]
```diff
@@ -0,0 +1,739 @@
...
+		}
+
+		return array_merge(
+			$categories,
+			array(
```
Comment (ErikIDX) [inline idx/blocks/register-blocks.php]: Should use `[]` syntax for consistency

File: idx/blocks/register-blocks.php [hunk 3]
```diff
@@ -0,0 +1,739 @@
...
+				],
+				'editor_script'   => 'impress-omnibar-block',
+				'render_callback' => [ $this, 'impress_omnibar_block_render' ],
+			]
+		);
+
+		$impress_omnibar_image_url = plugins_url( '/impress-omnibar/omnibar-placeholder.png', __FILE__ );
+		wp_localize_script( 'impress-omnibar-block', 'impress_omnibar_image_url', $impress_omnibar_image_url );
+		wp_enqueue_script( 'impress-omnibar-block' );
+
```
Comment (ErikIDX) [inline idx/blocks/register-blocks.php]: Extra line

File: idx/blocks/register-blocks.php [hunk 4]
```diff
@@ -0,0 +1,739 @@
...
+	 * Impress_lead_login_block_render function.
+	 *
+	 * @access public
+	 * @param mixed $attributes - Widget attributes.
+	 * @return string
+	 */
+	public function impress_lead_login_block_render( $attributes ) {
+		return $this->impress_shortcodes->lead_login_shortcode( $attributes );
+	}
+
```
Comment (ErikIDX) [inline idx/blocks/register-blocks.php]: Extra line

File: idx/blocks/register-blocks.php [hunk 5]
```diff
@@ -0,0 +1,739 @@
...
+	 *
+	 * @access public
+	 * @param mixed $attributes - Widget attributes.
+	 * @return string
+	 */
+	public function impress_omnibar_block_render( $attributes ) {
+		error_log( print_r( 'Omnibar render block called', true ) );
+		return $this->omnibar_shortcode->create_omnibar->add_omnibar_shortcode( $attributes );
+	}
+
```
Comment (ErikIDX) [inline idx/blocks/register-blocks.php]: Extra two lines

File: idx/blocks/register-blocks.php [hunk 6]
```diff
@@ -0,0 +1,739 @@
...
+			'idx-broker-platinum/idx-wrapper-tags-block',
+			[
+				'editor_script' => 'idx-wrapper-tags-block',
+			]
+		);
+
+		$idx_wrapper_tags_image_url = plugins_url( '/idx-wrapper-tags/wrapper-tag-placeholder.png', __FILE__ );
+		wp_localize_script( 'idx-wrapper-tags-block', 'idx_wrapper_tags_image_url', $idx_wrapper_tags_image_url );
+		wp_enqueue_script( 'idx-wrapper-tags-block' );
+
```
Comment (ErikIDX) [inline idx/blocks/register-blocks.php]: Extra line

File: idx/blocks/register-blocks.php [hunk 7]
```diff
@@ -0,0 +1,739 @@
...
+	 * Impress_carousel_block_render function.
+	 *
+	 * @access public
+	 * @param mixed $attributes - Widget attributes.
+	 * @return string
+	 */
+	public function impress_carousel_block_render( $attributes ) {
+		return $this->impress_shortcodes->property_carousel_shortcode( $attributes );
+	}
+
```
Comment (ErikIDX) [inline idx/blocks/register-blocks.php]: Extra line

File: idx/blocks/register-blocks.php [hunk 8]
```diff
@@ -0,0 +1,739 @@
...
+	 * Impress_showcase_block_render function.
+	 *
+	 * @access public
+	 * @param mixed $attributes - Widget attributes.
+	 * @return string
+	 */
+	public function impress_showcase_block_render( $attributes ) {
+		return $this->impress_shortcodes->property_showcase_shortcode( $attributes );
+	}
+
```
Comment (ErikIDX) [inline idx/blocks/register-blocks.php]: Extra line

File: idx/blocks/register-blocks.php [hunk 9]
```diff
@@ -0,0 +1,739 @@
...
+		wp_localize_script( 'impress-city-links-block', 'impress_city_links_block_image_url', $placeholder_image_url );
+
+		$mls_options = $this->get_mls_options();
+		wp_localize_script( 'impress-city-links-block', 'impress_city_links_mls_options', $mls_options );
+
+		$city_list_options = $this->get_city_list_options();
+		wp_localize_script( 'impress-city-links-block', 'impress_city_links_city_options', $city_list_options );
+
+		wp_enqueue_script( 'impress-city-links-block' );
+
```
Comment (ErikIDX) [inline idx/blocks/register-blocks.php]: Extra line

File: idx/blocks/register-blocks.php [hunk 10]
```diff
@@ -0,0 +1,739 @@
...
+	}
+
+	/**
+	 * Get_saved_links_list function.
+	 *
+	 * @access public
+	 * @return array
+	 */
+	public function get_saved_links_list() {
+
```
Comment (ErikIDX) [inline idx/blocks/register-blocks.php]: Extra line

File: idx/blocks/register-blocks.php [hunk 11]
```diff
@@ -0,0 +1,739 @@
...
+			}
+		}
+
+		if ( ! is_array( $saved_links ) ) {
+			return [];
+		}
+
+		return $saved_links_list;
+	}
+
```
Comment (ErikIDX) [inline idx/blocks/register-blocks.php]: Extra line

File: idx/blocks/register-blocks.php [hunk 12]
```diff
@@ -0,0 +1,739 @@
...
+
+	/**
+	 * Get_agents_select_list function.
+	 *
+	 * @access public
+	 * @return array
+	 */
+	public function get_agents_select_list() {
+
```
Comment (ErikIDX) [inline idx/blocks/register-blocks.php]: Extra line

File: idx/blocks/register-blocks.php [hunk 13]
```diff
@@ -0,0 +1,739 @@
...
+		$widget_list = [];
+
+		if ( $idx_widgets ) {
+			foreach ( $idx_widgets as $widget ) {
+				array_push( $widget_list, [ 'label' => $widget->name, 'value' => $widget->uid ] );
+			}
+		}
+		return $widget_list;
+	}
+
```
Comment (ErikIDX) [inline idx/blocks/register-blocks.php]: Extra line

File: idx/blocks/idx-widgets/script.js [hunk 14]
```diff
@@ -0,0 +1,59 @@
+(function (blocks, element, editor, components) {
+  var el = element.createElement
+  var registerBlockType = blocks.registerBlockType
+  var InspectorControls = editor.InspectorControls
+  var SelectControl = components.SelectControl
+  var icon = el('i', { class: 'fa fa-cog fa-2x' }, null)
+
+  function setCategory () {
+    if (window.location.href.includes('wp-admin')) {
```
Comment (ralls) [inline idx/blocks/idx-widgets/script.js]: This won't work with IE11 - need to use `indexOf`

**Pull request merged**